### PR TITLE
Support DL tag in TableReader brick

### PR DESCRIPTION
**Pull request by fregante**
- Fixes https://github.com/pixiebrix/pixiebrix-extension/issues/2831

## Live test area

<table id="support-table">
<tr>
	<th>
	<th>Chrome
	<th>Firefox
<tr>
	<th>Coolness
	<td>No
	<td>Yes
</table>

<dl id="definitions">
	<dt>Chrome</dt>
	<dd>Google’s browser</dd>
	<dt>Firefox</dt>
	<dd>Mozilla’s browser</dd>
</dl>

## Live test output

Side note: refer to https://github.com/pixiebrix/pixiebrix-extension/issues/2882 for table output expectations

Screen Shot 14

### CODE REVIEW

File: src/utils/parseDefinitionList.test.ts [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+  for (const [name, value] of Object.entries(attributes)) {
+    list.setAttribute(name, value);
+  }
+
+  return list;
+}
+
+describe("parseDefinitionList", () => {
+  test("parse simple list", () => {
+    const list = getDL(`
```
Comment (twschiller) [inline src/utils/parseDefinitionList.test.ts:37]: 🤣

Comment (fregante) [reply src/utils/parseDefinitionList.test.ts]: Urban Pixionary

**Review by twschiller — APPROVED** [review]
Learned a lot (of slang) from this PR

I'm a bit torn on if we should instead structuring it as having field names "term"/"detail". This would avoid concatenation for https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dl#single_term_multiple_descriptions. A term with multiple details would be multiple list items

Let's run with it. We can always expose an option later for the output shape

File: src/blocks/transformers/component/TableReader.ts [hunk 2]
```diff
@@ -81,8 +85,15 @@ export class TableReader extends Transformer {
   }
 
   async transform(args: BlockArg, { root }: BlockOptions): Promise<unknown> {
-    const $table = $safeFind<HTMLTableElement>(args.selector, root);
-    return parseDomTable($table.get(0), { orientation: args.orientation });
+    const table = requireSingleElement<HTMLTableElement | HTMLDListElement>(
```
Comment (fregante) [inline src/blocks/transformers/component/TableReader.ts:88]: I forgot to highlight this change: Until now the selector worked even if it returned multiple elements, e.g. `"table"`, but now that will throw errors. Is that too strict? Should it be changed to "require **at least** one element?

**Comment by fregante** [comment]
> Learned a lot (of slang) from this PR

```html
<dt>TWL
<dd>Today We Learned
```

> A term with multiple details would be multiple list items

Oh maybe I understood what you wanted:

- **DT: ciao**
- _DD: hello_
- _DD: bye_
- **DT: oh cielo**
- **DT: mamma mia**
- _DD: oh my god_

## This PR

<table>
<tr><th>
	<th>ciao
	<th>oh cielo
	<th>mamma mia
<tr><th>0
	<td>hello\nbye
	<td>oh my god
	<td>oh my god
</table>

## Wanted

<table>
<tr><th>
	<th>Term
	<th>Definition
<tr><th>0
	<td>ciao
	<td>hello
<tr><th>1
	<td>ciao
	<td>bye
<tr><th>2
	<td>oh cielo
	<td>oh my god
<tr><th>3
	<td>mamma mia
	<td>oh my god
</table>

**Comment by fregante** [comment]
I merged for now but please _review my "review" comment._

**Pull request merged**